### PR TITLE
Generalize `IsOurs` to work on different types of entity.

### PR DESCRIPTION
**Pull request by jonathanknowles**
# Issue Number

#899 

# Overview

This PR:

- [x] Generalizes the definition of the `IsOurs` type class so that it can be applied to more types of entities than Address.
- [x] Adds an `IsOurs s ChimericAccount` constraint to the `Wallet` type.

# Comments

For the moment, we provide an instance of `IsOurs (SeqState n k) ChimericAccount` that always returns `False`. A further PR will add this implementation.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs [hunk 1]
```diff
@@ -52,12 +54,12 @@ import Cardano.Wallet.Primitive.Types
 -- In practice, we will need a wallet that can support both, even if not at the
 -- same time, and this little abstraction can buy us this without introducing
 -- too much overhead.
-class IsOurs s where
+class IsOurs s entity where
```
Comment (rvl) [inline lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs:57]: I would still call this parameter `address`. `entity` is more general than it needs to be.

**Review by rvl — APPROVED** [review]
LGTM

File: lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs (hunk 1 above)
Comment (jonathanknowles) [reply lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs:57]: > I would still call this parameter `address`. `entity` is more general than it needs to be.

Ah, that naming suggestion came from a conversation with @KtorZ. The plan would be to eventually move this class to another module, to reflect the new level of generality. We would delay moving this class until a recovery week.

**Pull request merged**